x-pack/plugins/actions/server/lib/get_token_count_from_bedrock_invoke.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { encode } from 'gpt-tokenizer';
+
+export interface InvokeBody {
+  prompt: string;
+}
+
+/**
+ * Takes the Bedrock `run` and `test` sub action response and the request prompt as inputs.
+ * Uses gpt-tokenizer encoding to calculate the number of tokens in the prompt and completion.
+ * Returns an object containing the total, prompt, and completion token counts.
+ * @param response (string) - the response completion from the `run` or `test` sub action
+ * @param body - the stringified request prompt
+ */
+export async function getTokenCountFromBedrockInvoke({
+  response,
+  body,
+}: {
+  response: string;
+  body: string;
+}): Promise<{
+  total: number;
+  prompt: number;
+  completion: number;
+}> {
+  const chatCompletionRequest = JSON.parse(body) as InvokeBody;
+
+  // per https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb
+  const tokensFromMessages = encode(`<|start|>${chatCompletionRequest.prompt}<|end|>`).length;
+
+  const promptTokens = tokensFromMessages;
+
+  const completionTokens = encode(response).length;
```

Comment:
can we wrap these calls to `JSON.parse` and `encode` in a `try/catch`?

Reply:
Or when we call it [here](https://github.com/elastic/kibana/pull/170815/files#diff-7678f7e9330ad27120896f6a32fff43c1ad338f8a02eb2558f4f77ee320f7d65R97)